File: trading-ai-model/backend/api/routes/risk.py
"""Risk control API routes."""

from fastapi import APIRouter
from pydantic import BaseModel, Field

from risk.kill_switch_runtime import get_kill_switch_status, set_kill_switch_enabled
from risk.order_sizing_runtime import get_order_sizing, set_order_sizing

router = APIRouter()


class KillSwitchUpdate(BaseModel):
    enabled: bool


@router.get("/kill-switch")
def kill_switch_status():
    """Runtime kill switch state (Postgres + in-process override + env default)."""
    return get_kill_switch_status()


@router.put("/kill-switch")
async def update_kill_switch(body: KillSwitchUpdate):
    return await set_kill_switch_enabled(body.enabled)


class OrderSizingUpdate(BaseModel):
    coinbase_order_usd: float = Field(ge=1, le=10000)
    oanda_order_usd: float = Field(ge=1, le=10000)


@router.get("/order-sizing")
def order_sizing_status():
    return get_order_sizing()


@router.put("/order-sizing")
async def update_order_sizing(body: OrderSizingUpdate):
    return set_order_sizing(
        coinbase_order_usd=body.coinbase_order_usd,
        oanda_order_usd=body.oanda_order_usd,
    )

File: trading-ai-model/frontend/src/api/risk.js
const API_BASE = import.meta.env.VITE_API_URL ?? "/api";

export async function setKillSwitch(enabled) {
  const res = await fetch(`${API_BASE}/risk/kill-switch`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ enabled }),
  });
  if (!res.ok) {
    throw new Error(`Failed to update kill switch (${res.status})`);
  }
  return res.json();
}

export async function getOrderSizing() {
  const res = await fetch(`${API_BASE}/risk/order-sizing`);
  if (!res.ok) throw new Error(`Failed to load order sizing (${res.status})`);
  return res.json();
}

export async function setOrderSizing({ coinbase_order_usd, oanda_order_usd }) {
  const res = await fetch(`${API_BASE}/risk/order-sizing`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ coinbase_order_usd, oanda_order_usd }),
  });
  if (!res.ok) throw new Error(`Failed to update order sizing (${res.status})`);
  return res.json();
}
